### ISSUE TITLE
Setup Questions

### ISSUE THREAD

**Issue by muffin87** [contentful-labs/coredns-nodecache#21]
Hi,

first of all thank you! In our team we currently looking for a solution to lower
the amount of DNS queries to the central KubeDNS (we run CoreDNS). We tried the
official node-local-dns cache but we didn't like the idea to introduce a new
single point of failure.
Than we found your solution and the first read was awesome. We copied the
example deployment and changed the KubeDNS ip (since we have a different one).
Buuut the Daemonsets didn't start. We needed to add a ServiceAccount to the
deployment:

```yaml
apiVersion: v1
kind: ServiceAccount
metadata:
  name: coredns-nodecache
  namespace: kube-system
```

Than the primary Daemonset was starting. The next problem was that the
livenessProbe of the primary Pod was failing. Our question would be is the port
defined in the Daemonset (8080) the right one? Could that be that it's
port 8082? With that Port the probe was doing fine and all Pods of the pirmary
Daeomset are up and running.
The secondary still makes problems. The Kubelet says:

```
Events:
Type     Reason            Age        From               Message
----     ------            ----       ----               -------
Warning  FailedScheduling  <unknown>  default-scheduler  0/9 nodes are available: 6 node(s) didn't have free ports for the requested pod ports, 8 node(s) didn't match node selector.
```

That sounds like Kubernetes is not able to schedule a Pod (which runs in hostNetwork: true)
when there is already a Pod which binds to that port. Is there any trick oder
config we are missing?
Last question. In order to use the node cache what do we need to configure?
It's not working out of the box right? Do we need the reconfigure the Kubelets?
If yes do you add both IP addresses?

```yaml
# clusterDNS is a list of IP addresses for a cluster DNS server. If set,
# kubelet will configure all containers to use this for DNS resolution
# instead of the host's DNS servers.
clusterDNS:
  - <NODE LOCAL CACHE>
  - <KUBE DNS IP>
```

Is there a special treatment for the master nodes?

We running the following Setup:

+ OnPrem
+ Kubernetes 1.16.4
+ CentOS 7
+ Kernel 5.3.X

Thanks in advance
Cheers

**Comment by shaikatz** [comment]
I'm seeing the same 2 issues. using EKS 1.13. will be happy to get some assistance.

**Comment by yannh** [comment]
Thanks a lot for the feedback - it looks like we need to work on our documentation and fix some bugs in the provided manifests. 
Both pods should bind to the same IP address/port, you ll need to update the kubelet configuration in the same way you would for Kubernetes nodecache to use it. I will need to review why you are getting this error.
@shaikatz while we re not using EKS at the moment, I m very interested in getting it to work on EKS. I'll try to fix the manifests ASAP.

**Comment by argvk** [comment]
hey @muffin87 and @shaikatz, I just merged a PR to fix the manifests, would you mind trying with the updated manifests and letting us know ? 

> Do we need the reconfigure the Kubelets?

yes, you'd have set the clusterDNS to the value from https://github.com/contentful-labs/coredns-nodecache/blob/b195d73466562491f7236bed966338dcccdb9262/k8s/node-local-ha.yaml#L26

> If yes do you add both IP addresses?

We've set our clusterDNS to the node local cache ip alone.

**Comment by muffin87** [comment]
Hi @argvk and @yannh,

thanks for the quick response :)!
Now it's working fine.

For me changing the Prometheus metric ports did the trick. They needed to be different for the primary and secondary. I deployed it a few hours ago in our development cluster. Works like a charm.

The only thing i changed was to make sure that the Daemonset is also deployed on
the Master nodes since this makes the Kubelet config easier. I only needed to
add the CoreDNS nodecache IP.

I also tried to kill / break the DNS and even when I delete both Pods on the
node I only have a few seconds no DNS service. So... great job! I really like it.
I also look up the metrics and the improvement is HUGE! Without the cache I 
had 6 to 7 seconds response times in the 99.999% quantil and with the cache I'm
down to 100ms. 

Cheers
